### ISSUE TITLE
Consider deprecating setSurfaceSize

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#37626]
You can do the same with `physicalSizeTestValue`.

History: "setSurfaceSize was first IIRC, and physicalSizeTestValue was second. The first solved the specific problem of setting the size of the test surface. The second allowed you to just override anything on the Window."

**Comment by dnfield** [comment]
Is it an API break if we just mark it deprecated?

**Comment by goderbauer** [comment]
Our analyzer tests would fail if we'd continue to use it, so for us it would be :) . And the goal is to eventually remove it altogether. So at that point is would be an API break.

**Comment by dnfield** [comment]
Yes, but we could make it a "soft" break by just deprecating it before we remove it, with the idea being that analysis failures are a softer break than just deleting the API.

**Comment by goderbauer** [comment]
Yeah, we should do that. Keep it deprecated for a couple of Flutter releases and then just remove it.
